src/main/kotlin/com/autonomousapps/tasks/ComputeDominatorTreeTask.kt

```diff
@@ -153,8 +166,19 @@ abstract class ComputeDominatorTreeTask : DefaultTask() {
         nodeWriter = nodeWriter,
       )
 
+      val dataWriter = DominanceTreeDataWriter(
```

Comment:
Should this only be done if a specific option is set to avoid any perf penalties for consumers which don't care about a JSON output?

Reply:
No, this is fine.